test/entities/swapV3.integration.test.ts

```diff
@@ -0,0 +1,335 @@
+        });
+    });
+    describe('swap should be executed correcly', () => {
+        describe('wethIsEth: false', () => {
+            const swapParams = {
+                chainId,
+                paths: [pathBalWeth],
+                wethIsEth: false,
+            };
+            test('GivenIn', async () => {
+                const swap = new Swap({
+                    ...swapParams,
+                    swapKind: SwapKind.GivenIn,
+                });
+                await assertSwapExactIn(client, chainId, swap, false);
+            });
+            test('GivenOut', async () => {
+                const swap = new Swap({
+                    ...swapParams,
+                    swapKind: SwapKind.GivenOut,
+                });
+                await assertSwapExactOut(client, chainId, swap, false);
+            });
+        });
+        describe('wethIsEth: true', () => {
+            describe('eth out', async () => {
+                test('GivenIn', async () => {
+                    const swap = new Swap({
+                        chainId,
+                        paths: [pathBalWeth],
+                        swapKind: SwapKind.GivenIn,
+                        wethIsEth: true,
+                    });
+                    await assertSwapExactIn(client, chainId, swap, true);
+                });
+                test('GivenOut', async () => {
+                    const swap = new Swap({
+                        chainId,
+                        paths: [pathBalWeth],
+                        swapKind: SwapKind.GivenOut,
+                        wethIsEth: true,
+                    });
+                    await assertSwapExactOut(client, chainId, swap, true);
+                });
+            });
+            describe('eth in', () => {
+                test('GivenIn', async () => {
+                    const pathWethBal = {
+                        ...pathBalWeth,
+                        tokens: [...pathBalWeth.tokens].reverse(),
+                    };
+                    const swap = new Swap({
+                        chainId,
+                        paths: [pathWethBal],
+                        swapKind: SwapKind.GivenIn,
+                        wethIsEth: true,
+                    });
+                    await assertSwapExactIn(client, chainId, swap, true);
+                });
+                test('GivenOut', async () => {
+                    const pathWethBal = {
+                        ...pathBalWeth,
+                        tokens: [...pathBalWeth.tokens].reverse(),
+                    };
+                    const swap = new Swap({
+                        chainId,
+                        paths: [pathWethBal],
+                        swapKind: SwapKind.GivenOut,
+                        wethIsEth: true,
+                    });
+                    await assertSwapExactOut(client, chainId, swap, true);
+                });
+            });
+        });
+    });
+});
+
+async function assertSwapExactIn(
```

Comment:
Would be good to move this function to a different file on the folder `./test/lib/utils`, just as another test helper functions.

Reply:
I think its better to keep it here for now as its not being used anywhere else. Might change after batchSwaps are added though.